src/postgres/services/PostgresCodeLensProvider.ts

```diff
@@ -0,0 +1,66 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { CodeLens, CodeLensProvider, Event, EventEmitter, Position, ProviderResult, Range } from "vscode";
+import { callWithTelemetryAndErrorHandling, IActionContext } from "vscode-azureextensionui";
+import { localize } from "../../utils/localize";
+
+export class PostgresCodeLensProvider implements CodeLensProvider {
+    private _onDidChangeEmitter: EventEmitter<void> = new EventEmitter<void>();
+    private _connectedDatabase: string | undefined;
+    private _connectedDatabaseInitialized: boolean;
+
+    public get onDidChangeCodeLenses(): Event<void> {
+        return this._onDidChangeEmitter.event;
+    }
+
+    public setConnectedDatabase(database: string | undefined): void {
+        this._connectedDatabase = database;
+        this._connectedDatabaseInitialized = true;
+        this._onDidChangeEmitter.fire();
+    }
+
+    public provideCodeLenses(): ProviderResult<CodeLens[]> {
```

Comment:
Just out of curiosity, what did you use as a reference for this?  I just want to know since I don't have a lot of knowledge on how to providing code lenses.

Reply:
I modeled it after the mongo implementation: https://github.com/microsoft/vscode-cosmosdb/blob/master/src/mongo/services/MongoCodeLensProvider.ts